packages/dashboard/test/dashboard-layout.test.ts

```diff
@@ -613,3 +616,73 @@ describe('dashboard layout', () => {
     });
   });
 });
+
+describe('root heading level', () => {
+  let dashboard: DashboardLayout;
+  let section: DashboardSection;
+  let widget: DashboardWidget;
+  let nestedWidget: DashboardWidget;
+
+  beforeEach(async () => {
+    dashboard = fixtureSync(`
+      <vaadin-dashboard-layout>
+        <vaadin-dashboard-widget widget-title="Widget"></vaadin-dashboard-widget>
+        <vaadin-dashboard-section section-title="Section">
+          <vaadin-dashboard-widget widget-title="Nested Widget"></vaadin-dashboard-widget>
+        </vaadin-dashboard-section>
+      </vaadin-dashboard-layout>
+    `);
+    await nextFrame();
+    await nextResize(dashboard);
+    widget = dashboard.querySelector('vaadin-dashboard-widget') as DashboardWidget;
+    section = dashboard.querySelector('vaadin-dashboard-section') as DashboardSection;
+    nestedWidget = section.querySelector('vaadin-dashboard-widget') as DashboardWidget;
+  });
+
+  function assertHeadingLevels(expectedHeadingLevel: number) {
+    assertHeadingLevel(widget, expectedHeadingLevel);
+    assertHeadingLevel(section, expectedHeadingLevel);
+    assertHeadingLevel(nestedWidget, expectedHeadingLevel + 1);
+  }
+
+  it('should use default title heading level (2) when not explicitly set', () => {
+    assertHeadingLevels(2);
+  });
+
+  it('should use custom title heading level when set on dashboard layout', async () => {
+    dashboard.rootHeadingLevel = 4;
+    await nextFrame();
+    assertHeadingLevels(4);
+  });
+
+  it('should update title heading level when changed on dashboard layout', async () => {
```

Comment:
This seems superfluous with the test above.

Reply:
Removed duplicate tests.